### PR TITLE
feat: remove overwrite apisix.yaml

### PR DESCRIPTION
**Pull request by kamly**
fix https://github.com/apache/apisix-docker/issues/482

### CODE REVIEW

**Review by nfrankel — APPROVED** [review]
Thanks for your contribution!

**Comment by kamly** [comment]
@soulbird hi, please have a review ~

**Review by Sn0rt — APPROVED** [review]

**Comment by kamly** [comment]
@nfrankel @Sn0rt , hi I meet two problems

1. if remove overwrite `apisix.yaml`

when docker run with `-e APISIX_STAND_ALONE=true`
```shell
docker run -d --name apache-apisix \
  -p 9080:9080 \
  -e APISIX_STAND_ALONE=true \
  apache/apisix
```

will print warn and error info
```
...
apisix              | 2023/07/31 12:55:13 [warn] 47#47: *11206 [lua] config_yaml.lua:127: wait for more time, context: ngx.timer
apisix              | 2023/07/31 12:55:13 [error] 31#31: *12907 [lua] config_yaml.lua:73: failed to fetch /usr/local/apisix/conf/apisix.yaml attributes: cannot obtain information from file '/usr/local/apisix/conf/apisix.yaml': No such file or directory, context: ngx.timer
...
```

is it as expected ?

2. Unit tests fail. (https://github.com/apache/apisix-docker/actions/runs/5689010648/job/15470480442)

I debugged the unit test and found that it will exit automatically. So executing `curl http://127.0.0.1:9000` will fail

image

**Review by nfrankel — CHANGES_REQUESTED** [review]
>is it as expected ?

Nope, it's not. Apache APISIX requires the file. So the correct logic must be that **if** the file exists, don't overwrite it; if it does, keep the current behavior

**Comment by kamly** [comment]
> Nope, it's not. Apache APISIX requires the file. So the correct logic must be that **if** the file exists, don't overwrite it; if it does, keep the current behavior

done

**Comment by kamly** [comment]
@nfrankel please , look look ~

**Review by nfrankel — APPROVED** [review]

**Comment by nfrankel** [comment]
> @nfrankel others can review ?

We appreciate your contribution, but please, don't comment to get faster review. People do have other commitments. Hence:

1. It doesn't speed up things
2. It makes people unhappy

Thanks for your understanding.

Besides, one of the check is failing.

**Comment by kamly** [comment]
> Besides, one of the check is failing.

got it ,  I just want to push on and see what else I need to do

**Comment by nfrankel** [comment]
>got it , I just want to push on and see what else I need to do

I appreciate your willingness to move forward and understand your eagerness to do more. Unfortunately, at the moment, all reviewers are busy.

As for me, I'm now on vacations. Please be patient.

File: centos/docker-entrypoint.sh [hunk 1]
```diff
@@ -29,11 +29,13 @@ deployment:
     config_provider: yaml
 _EOC_
 
+    if [ ! -f "${PREFIX}/conf/apisix.yaml" ]; then
         cat > ${PREFIX}/conf/apisix.yaml << _EOC_
 routes:
   -
 #END
 _EOC_
+    fi
```
Comment (tao12345666333) [inline centos/docker-entrypoint.sh:38]: The indentation here is incorrect.

**Review by tao12345666333 — CHANGES_REQUESTED** [review]

**Comment by kamly** [comment]
@tao12345666333 fix~

**Comment by Revolyssup** [comment]
@kamly one test still failing

**Comment by kamly** [comment]
> @kamly one test still failing

I debugged the unit test and found that it will exit automatically. So executing curl http://127.0.0.1:9000/ will fail

image

I need help

**Comment by kamly** [comment]
> @kamly one test still failing

fix it ~

**Comment by moonming** [comment]
@tao12345666333 @Revolyssup please take a look this PR, thx

File: centos/docker-entrypoint.sh [hunk 2]
```diff
@@ -29,11 +29,13 @@ deployment:
     config_provider: yaml
 _EOC_
 
-        cat > ${PREFIX}/conf/apisix.yaml << _EOC_
+        if [ ! -f "${PREFIX}/conf/apisix.yaml" ]; then
```
Comment (Revolyssup) [inline centos/docker-entrypoint.sh:32]: [This](https://github.com/apache/apisix-docker/pull/487/files#diff-8c251a9432dc64cc4a02088f433622d5bd6833832d0214fe045623021b226b92R25)  line is not protected by the if block so there will be override anyways. Shouldn't this if block moved upwards to protect the whole block?

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: centos/docker-entrypoint.sh (hunk 2 above)
Comment (Revolyssup) [reply centos/docker-entrypoint.sh:32]: cc: @kamly

**Review by tao12345666333 — APPROVED** [review]
LGTM

File: centos/docker-entrypoint.sh (hunk 2 above)
Comment (kamly) [reply centos/docker-entrypoint.sh:32]: > [This](https://github.com/apache/apisix-docker/pull/487/files#diff-8c251a9432dc64cc4a02088f433622d5bd6833832d0214fe045623021b226b92R25) line is not protected by the if block so there will be override anyways. Shouldn't this if block moved upwards to protect the whole block?

Do you mean: Can `conf/config.yaml` be overwritten ?

In my opinion, if `$APISIX_STAND_ALONE==true` is true, the content of `conf/config.yaml` should be reset

conf/config.yaml
```
deployment:
  role: data_plane	  
  role_data_plane:
    config_provider: yaml
```

Comment (Revolyssup) [reply centos/docker-entrypoint.sh:32]: [This](https://github.com/apache/apisix-docker/blob/126d19e3e1e6dad3467f9e815dcdace4d64a4cc0/debian/docker-entrypoint.sh#L32) is also being executed when `$APISIX_STAND_ALONE==true`.
From [here](https://github.com/apache/apisix-docker/blob/126d19e3e1e6dad3467f9e815dcdace4d64a4cc0/debian/docker-entrypoint.sh#L25) to [here](https://github.com/apache/apisix-docker/blob/126d19e3e1e6dad3467f9e815dcdace4d64a4cc0/debian/docker-entrypoint.sh#L36) is part of the same branch

Comment (kamly) [reply centos/docker-entrypoint.sh:32]: @Revolyssup I'm sorry, I don't quite understand what you mean, can you provide a demo?

my logic is:
when `$APISIX_STAND_ALONE==true` overwrite `conf/config.yaml`
then `! -f "${PREFIX}/conf/apisix.yaml"` is true overwrite `conf/apisix.yaml`

Only standalone need `conf/apisix.yaml`

Comment (kamly) [reply centos/docker-entrypoint.sh:32]: cc @tao12345666333 , I need help

Comment (kamly) [reply centos/docker-entrypoint.sh:32]: @Revolyssup hi?

**Comment by Revolyssup** [comment]
@kamly Why did you close it? 
My query was that this whole block is one if branch
``` bash
    if [ "$APISIX_STAND_ALONE" = "true" ]; then
        cat > ${PREFIX}/conf/config.yaml << _EOC_
deployment:
  role: data_plane
  role_data_plane:
    config_provider: yaml
_EOC_

        cat > ${PREFIX}/conf/apisix.yaml << _EOC_
routes:
  -
#END
_EOC_
        /usr/bin/apisix init
```

But we were only add the extra protection on this code
``` bash
        cat > ${PREFIX}/conf/apisix.yaml << _EOC_
routes:
  -
#END
_EOC_
```

and not this
``` bash
        cat > ${PREFIX}/conf/config.yaml << _EOC_
deployment:
  role: data_plane
  role_data_plane:
    config_provider: yaml
_EOC_
```

cc: @tao12345666333

**Comment by kamly** [comment]
I made a mistake, I resubmit the PR

**Pull request closed without merging**